spec/unit/loader_spec.rb

```diff
@@ -117,4 +117,44 @@ module Test
       it_behaves_like "object loader"
     end
   end
+
+  describe "#constant" do
+    let(:component) {
+      Dry::System::Component.new(
+        Dry::System::Identifier.new("test.api_bar"),
+        namespace: Dry::System::Config::Namespace.default_root,
+        inflector: Dry::Inflector.new { |i| i.acronym("API") }
+      )
+    }
+    describe "successful constant loading" do
+      before do
+        Test::APIBar = Class.new
+      end
+
+      it "returns the constant" do
+        expect(loader.constant(component)).to eq(Test::APIBar)
+      end
+    end
+
+    describe "unsuccessful constant loading" do
+      before do
+        Test::APIBoo = Class.new
+      end
+
+      it "raises custom error" do
+        expect { loader.constant(component) }.to raise_error(
+          Dry::System::ComponentNotLoadableError
+        ) do |error|
+          expect(error.message).to eq(
```

Comment:
Should I get rid of this message expectation? We already test the message in the new `spec/unit/dry/system/errors_spec.rb` file